# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GC.cpp
# Commit: 27d9b7db5e1c
# Full Hash: 27d9b7db5e1c50af9456023d63e2d3db72b48df2
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-10-28 04:43:55
# Regressor Bug: 1797755
# File Overlap Count: 1
# Description:
#   Bug 1797755 - Part 5: Use a single initial mark stack size regardless of whether incremental GC is enabled r=sfink
#   
#   Currently we initialize the mark stack to a different size depending on whether
#   or not incremental GC is enable. However, after the first GC we always shrink
#   it to the initial size when it is disabled.
# ==============================================================================

diff -r f207beed6252 -r 27d9b7db5e1c js/src/gc/GC.cpp
--- a/js/src/gc/GC.cpp	Thu Oct 27 18:23:03 2022 +0000
+++ b/js/src/gc/GC.cpp	Thu Oct 27 18:23:03 2022 +0000
@@ -465,7 +465,6 @@
       nursery_(this),
       storeBuffer_(rt, nursery()),
       lastAllocRateUpdateTime(TimeStamp::Now()) {
-  marker.setIncrementalGCEnabled(incrementalGCEnabled);
 }
 
 using CharRange = mozilla::Range<const char>;
@@ -1231,7 +1230,6 @@
 
 void GCRuntime::setIncrementalGCEnabled(bool enabled) {
   incrementalGCEnabled = enabled;
-  marker.setIncrementalGCEnabled(enabled);
 }
 
 void GCRuntime::updateHelperThreadCount() {